visualization/app/codeCharta/services/loadInitialFile/loadInitialFile.service.ts

```diff
@@ -25,50 +71,351 @@ export class LoadInitialFileService {
-			const message = "One or more files from the given file URL parameter could not be loaded. Loading sample files instead."
-			let title = "Error"
-			if (error.message) {
-				title += ` (${error.message})`
-			} else if (error.statusText && error.status) {
-				title += ` (${error.status}: ${error.statusText})`
+	private async handleErrorLoadFilesFromQueryParams(error: Error) {
+		if ((error as Error).message !== NO_FILES_LOADED_ERROR_MESSAGE) {
+			const title = "File(s) could not be loaded from the given file URL parameter. Loaded sample files instead."
+			const message = this.createTitleUrlErrorDialog(error as Error)
+			this.showErrorDialog(title, message)
+		}
+		await this.loadSampleFiles()
+	}
+
+	private async loadFilesFromIndexedDB() {
+		try {
+			const savedCcState = await readCcState()
+			if (!savedCcState) {
+				await this.loadSampleFiles()
+				return
 			}
-			this.dialog.open(ErrorDialogComponent, {
-				data: { title, message }
-			})
+
+			const savedFileStates = savedCcState.files
+			const savedNameDataPairs = savedFileStates.map(fileState => getNameDataPair(fileState.file))
+			this.applyAppSettings(savedCcState.appSettings)
+			this.loadFileService.loadFiles(savedNameDataPairs)
+			this.store.dispatch(setFiles({ value: savedFileStates }))
+			this.applyAllSettings(savedCcState)
+		} catch (error) {
+			await this.handleErrorLoadFilesFromIndexedDB(error as Error)
+		}
+	}
+
+	private async handleErrorLoadFilesFromIndexedDB(error: Error) {
+		if ((error as Error).message !== NO_FILES_LOADED_ERROR_MESSAGE) {
+			const title = "File(s) could not be loaded from indexeddb. Loaded sample files instead."
+			const message = (error as Error).message
+			this.showErrorDialog(title, message)
 		}
-		this.tryLoadingFiles([
-			{ fileName: "sample1.cc.json", fileSize: 3 * 1024, content: sample1 as ExportCCFile },
-			{ fileName: "sample2.cc.json", fileSize: 2 * 1024, content: sample2 as ExportCCFile }
-		])
+		await this.loadSampleFiles()
 	}
 
-	private tryLoadingFiles(values: NameDataPair[]) {
-		GlobalSettingsHelper.setGlobalSettingsOfLocalStorageIfExists(this.store, this.state.getValue().appSettings)
-		this.loadFileService.loadFiles(values)
+	private async applyAllSettings(savedCcState: CcState) {
+		const savedFileSettings = savedCcState.fileSettings
+		const savedDynamicSettings = savedCcState.dynamicSettings
+		const savedAppSettings = savedCcState.appSettings
+		const missingPropertiesInSavedCcState = []
+		if (savedFileSettings) {
+			const missingFileSettings = this.applyFileSettings(savedFileSettings)
+			missingPropertiesInSavedCcState.push(...missingFileSettings)
+		}
+		if (savedDynamicSettings) {
+			const missingDynamicSettings = this.applyDynamicSettings(savedDynamicSettings)
+			missingPropertiesInSavedCcState.push(...missingDynamicSettings)
+		}
+		if (savedAppSettings) {
+			const missingAppSettings = this.applyAppSettings(savedAppSettings)
+			missingPropertiesInSavedCcState.push(...missingAppSettings)
```

Comment:
If something is not in the savedState, it is missing... ok. But shouldn't we simply add the missing properties to the savedState, so that it is up to date instead of showing an error? Maybe I did not understand it yet.

Reply:
I implemented error-handling to ensure changes in the format of the cc.json or our internal settings are handled gracefully. If you've previously blacklisted items in CodeCharta and we've since updated how we store those settings, there will be a notification informing you that all settings were restored correctly, except the blacklisted items (as you would have expected).